src/merkle/partial_mt/mod.rs

```diff
@@ -160,6 +164,26 @@ impl PartialMerkleTree {
         })
     }
 
+    /// Returns an iterator over the inner nodes of this Merkle tree.
+    pub fn inner_nodes(&self) -> impl Iterator<Item = InnerNodeInfo> + '_ {
+        let inner_nodes = self.nodes.iter().filter(|(index, _)| !self.leaves.contains(index));
+        inner_nodes.map(|(index, digest)| {
+            let left_index = NodeIndex::new(index.depth() + 1, index.value() * 2)
+                .expect("Failure to get left child index");
+            let right_index = NodeIndex::new(index.depth() + 1, index.value() * 2 + 1)
+                .expect("Failure to get right child index");
+            let left_hash =
+                self.nodes.get(&left_index).cloned().expect("Failure to get left child hash");
+            let right_hash =
+                self.nodes.get(&right_index).cloned().expect("Failure to get right child hash");
```

Comment:
minor nits: I'd say "Failed to get ..." instead of "Failure to get ...". Also, cloning could be done later on lines 181 and 182.